data_management/access_copy_converter/access_copy_converter/main.py

```diff
@@ -60,106 +70,169 @@ def get_extension(file_id, conn, metadata):
+                input_path = os.path.join(tmpdir, f"input.{extension}")
+                output_path = os.path.join(tmpdir, "input.pdf")
+
+                s3.download_file(source_bucket, key, input_path)
+                logger.info(f"Downloaded {key} to {input_path}")
+
+                try:
+                    if extension in ("xls", "xlsx"):
+                        temp_ods = os.path.join(tmpdir, "input.ods")
+                        convert_with_libreoffice(
+                            input_path, temp_ods, convert_to="ods"
+                        )
+                        convert_with_libreoffice(
+                            temp_ods,
+                            output_path,
+                            convert_to='pdf:calc_pdf_Export:{"SinglePageSheets":{"type":"boolean","value":"true"}}',
+                        )
+                    else:
+                        convert_with_libreoffice(input_path, output_path)
+                    logger.info(f"Converted {input_path} to PDF {output_path}")
+                except Exception as e:
+                    logger.error(f"Conversion failed for {file_id}: {e}")
+                    continue
 
-                with tempfile.TemporaryDirectory() as tmpdir:
-                    input_path = os.path.join(tmpdir, f"input.{extension}")
-                    output_path = os.path.join(tmpdir, "input.pdf")
+                s3.upload_file(output_path, dest_bucket, dest_key)
+                logger.info(
+                    f"Uploaded converted file {file_id} to Access Copy Bucket"
+                )
 
-                    s3.download_file(source_bucket, key, input_path)
-                    logger.info(f"Downloaded {key} to {input_path}")
 
-                    try:
-                        convert_office_to_pdf(input_path, output_path)
-                        logger.info(
-                            f"Converted {input_path} to PDF {output_path}"
-                        )
-                    except Exception as e:
-                        logger.error(f"Conversion failed for {file_id}: {e}")
-                        continue
-
-                    dest_key = f"{consignment_ref}/{file_id}"
-                    s3.upload_file(output_path, dest_bucket, dest_key)
-                    logger.info("Uploaded converted PDF to Access Copy Bucket")
-            else:
-                logger.info(f"File {file_id} does not require conversion")
+def create_access_copies_for_all_consignments(source_bucket):
+    paginator = s3.get_paginator("list_objects_v2")
+    response = paginator.paginate(Bucket=source_bucket)
+    consignments = set()
+    for page in response:
+        for obj in page["Contents"]:
+            consignments.add(obj["Key"].split("/")[0])
 
+    logger.info(f"Found {len(consignments)} consignments")
 
-def main():
+    for consignment_ref in consignments:
+        return consignment_ref
+
+
+def create_access_copy_from_sns():
```

Comment:
this function does not create any access copies, it just returns the consignment_reg mentioned in the sns message